src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const stringToObject = sourceString
+    .split(';')
+    .map((value) => {
+      return value.split(':').map((spaces) => spaces.trim());
+    })
+    .reduce((acc, key) => {
+      acc[key[0]] = key[1];
+
+      return acc;
+    }, {});
```

Comment:
The 'reduce' callback function should check if 'key[0]' is not an empty string before adding it to the accumulator object to avoid adding unwanted properties. Also, ensure that 'key[1]' is defined before assigning it to 'acc[key[0]]'.